protos/CMakeLists.txt

```diff
@@ -63,6 +64,9 @@ set_target_properties(
       EXPORT_COMPILE_COMMANDS OFF
 )
 
+configure_codecov(srf_protos)
+configure_codecov(srf_architect_protos)
+
```

Comment:
There is no reason not to do it; and I'd argue that this isn't true in the general case. If we have protobuf code that is never run, it can indicate code rot, or unintended behavior.

Reply:
That said, I have it disabled in the coverage calculation at the moment, so its not hurting stats.

Reply:
> That said, I have it disabled in the coverage calculation at the moment, so its not hurting stats.

Thats good, but I am still hesitant. The proto generation stuff can be very fickle. And just because some of our tests dont hit some of the protos, that doesnt mean it's not used. There could be some external tool that needs it, or bindings in another language that use the code. And a huge majority of the proto code is just never going to be used.

Reply:
`append_coverage_compiler_flags()` will only alter targets created after this function is called. Same with the changes made in `include(cmake/setup_compiler.cmake)`. So the order is very important here:

1. Setup options for all targets, including dependencies
   1. Do things like setting up the cache, finding CUDA, etc.
3. Configure dependencies
4. Setup options for all SRF targets, including protos
   1. Set the default library suffix and enable exporting the compile commands.
   2. We dont need things like full debugging and clang-tidy for these targets because we have no control over the generated code.
6. Configure protos
7. Setup options for all, non-generated, SRF targets
   1. Set up the compilers for all code that we have control over
      1. Things like setting the debug optimization thats best for setting breakpoints, adding clang-tidy/iwyu
10. Configure SRF targets

Thats why I think `append_coverage_compiler_flags()` should happen in step 5 and the protos should be moved back to where they were

Reply:
Done